RFC/src/RFC-0171_PeerToPeerMessagingProtocol.md

```diff
@@ -0,0 +1,662 @@
+| -------- | ----- | --------------- | --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| identity | 0     | 8               | `[u8;8]`  | The identifier that a `ZMQ_ROUTER` socket expects so that it knows the intended destination of the message. This can be thought of as a session token.          |
+| version  | 1     | 1               | `u8`      | The wire protocol version.                                                                                                                                      |
+| header   | 2     | as required     | `Vec<u8>` | Serialized bytes of data containing an unencrypted HeaderMessage.                                                                                               |
+| body     | 3     | as required     | `Vec<u8>` | Serialized bytes of data containing a unencrypted or encrypted message.                                                                                         |
+
+The header and decrypted body MUST be deserializable as per [RFC-0171: Message Serialisation](RFC-0171_MessageSerialisation.md)
+
+#### MessageHeader
+
+Every Tari message must have a header containing the following fields:
+
+| Name         | Type              | Description                                                                                        |
+| ------------ | ----------------- | -------------------------------------------------------------------------------------------------- |
+| version      | `String`          | The message protocol version (e.g. "Tari/1.0")."                                                   |
+| message_type | `MessageType`     | An enumeration of the message type of the body. As per [MessageType].                              |
+| source       | `PublicKey`       | The source public key.                                                                             |
+| dest         | `Option<NodeId>` | The destination [node ID]. This is optional.                                                       |
+| signature    | `[u8]`         | Signature of the message header and body, signed with the private key of the source.               |
+| flags        | `u8`              | <ul><li>bit 0: 1 indicates that the message body is encrypted</li><li>bits 1-7: reserved</li></ul> |
+
+A [communication node] and [communication client]:
+
+- MUST validate the signature of the message using the source's public key.
+- MUST reject the message if the signature verification fails.
+- if the encryption bit flag is set:
+    - MUST attempt to decrypt the message, or failing that
+    - MUST forward the message to a subset of peers using the `Closest` [BroadcastStrategy]
+    - MUST discard the message if the body is not encrypted
+    
+#### Messages
+
+Messages are an intention to perform a task, and so should be a verb like `Join` or `AddBlock`.
```

Comment:
I think so, or something like `chat::SendMsg` - I will rework this concept a bit because I thought of this as a string representation of message type, not integer. So how MessageType are mapped  from `u8` becomes more an implementation detail and the namespace concept kind of falls away

Reply:
Right, I interpreted this as the Rust enum naming convention